core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,226 @@
+package cats
+package data
+
+import scala.annotation.tailrec
+import scala.collection.immutable.VectorBuilder
+import cats.instances.vector._
+
+/**
+ * A data type which represents a non empty Vector.
+ */
+final case class NonEmptyVector[A] private (toVector: Vector[A]) {
+
+  /** Gets the element at the index, if it exists */
+  def get(i: Int): Option[A] =
+    toVector.lift(i)
+
+  /** Gets the element at the index, or throws an exception if none exists */
+  def getUnsafe(i: Int): A = toVector(i)
+
+  /** Updates the element at the index, if it exists */
+  def updated(i: Int, a: A): Option[NonEmptyVector[A]] =
+    if (toVector.isDefinedAt(i)) Some(NonEmptyVector(toVector.updated(i, a))) else None
+
+  /** Updates the element at the index, or throws an exeption if none exists */
+  def updatedUnsafe(i: Int, a: A):
+      NonEmptyVector[A] = NonEmptyVector(toVector.updated(i, a))
+
+  def head: A = toVector.head
+
+  def tail: Vector[A] = toVector.tail
+
+  /**
+   * remove elements not matching the predicate
+   */
+  def filter(f: A => Boolean): Vector[A] = toVector.filter(f)
+
+  /**
+   * Append another NonEmptyVector to this
+   */
+  def concat(other: NonEmptyVector[A]): NonEmptyVector[A] = NonEmptyVector(toVector ++ other.toVector)
+
+  /**
+    * Alias for concat
+    */
+  def ++(other: NonEmptyVector[A]): NonEmptyVector[A] = concat(other)
+
+  /**
+   * Append another Vector to this
+   */
+  def concat(other: Vector[A]): NonEmptyVector[A] = NonEmptyVector(toVector ++ other)
+
+  /**
+    * Alias for concat
+    */
+  def ++(other: Vector[A]): NonEmptyVector[A] = concat(other)
```

Comment:
Again some minor nitpicking about formatting, but both the `++` methods ScalaDoc's can be changed to 

```scala
/**
 * Alias for concat
 */
```